### PR TITLE
Add support for AltvMP

### PR DESCRIPTION
**Pull request by xCausxn**
This PR adds support for AltvMP. (#587)

It implements two methods to find the server using the master list at https://api.alt-mp.com/servers

If the `serverId` argument is passed, then a direct lookup can be done at /servers/ID

This does also add a fix to state.connect in general, as if the connect field is manually set, core.js just overrides it. (likely something that was missed when refactored)

### CODE REVIEW

**Comment by xCausxn** [comment]
@CosminPerRam a potential implementation that adds some flexibility. 

Seeing as we do `guildId` for discord, seemed like an ok addition. We could even just look to change `guildId` to also be `serverId` (with backwards compat of course) as its a bit more generic and could apply to multiple servers implementations in the future.

**Comment by CosminPerRam** [comment]
Looks great, I really like the option to use the id or the host:port but could you argue the addition of the `useCdn` option? In [their docs](https://docs.altv.mp/articles/cdn_links.html), its just a CDN, so it *could* be used for multiple servers, right? I haven't checked with a script to see if there are multiple entries with the same cdn url, but could it be possible to do so?

File: protocols/core.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ export default class Core extends EventEmitter {
     state.queryPort = options.port
     // because lots of servers prefix with spaces to try to appear first
     state.name = (state.name || '').trim()
-    state.connect = `${state.gameHost || options.host || options.address}:${state.gamePort || options.port}`
+    state.connect = state.connect || `${state.gameHost || options.host || options.address}:${state.gamePort || options.port}`
```
Comment (CosminPerRam) [inline protocols/core.js:87]: This is technically a breaking change and a bug, currently `discord`, `eldewrito` and `buildandshoot` do (or may) set this field manually and although the state should use that one, it does not and manually sets one (via the fields before editing).

It is a breaking change as users of these protocols could get different data after updating to a newer version.
And it is a bug as the field should respect the protocol's selection and not overwrite this.

**Comment by CosminPerRam** [comment]
> Seeing as we do guildId for discord, seemed like an ok addition. We could even just look to change guildId to also be serverId (with backwards compat of course) as its a bit more generic and could apply to multiple servers implementations in the future.

For sure, great idea, please make a tracking issue for this (:

File: protocols/core.js (hunk 1 above)
Comment (CosminPerRam) [reply protocols/core.js:87]: Talked over on discord and this should be fine.

**Review by CosminPerRam — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 ## To Be Released...
 ## 5.X.Y
 * FOUNDRY - Added support (#585)
+* Grand Theft Auto V: AltVMP - Added support (By @xCausxn #588)
```
Comment (CosminPerRam) [inline CHANGELOG.md:6]: Actually, please mention the `state.connect` as a fix here.

**Pull request merged**